In [1]:
import pandas as pd
import numpy as np
#import seaborn as sns
#import matplotlib.pyplot as plt
import scipy.stats as scp
#import statsmodels.formula.api as smf
#import statsmodels.api as sm

In [2]:
data=pd.read_csv('Advertising.csv')

# 1. Explain the dataset

In [3]:
data.head()

Unnamed: 0     TV  Radio  Newspaper  Sales
0           1  230.1   37.8       69.2   22.1
1           2   44.5   39.3       45.1   10.4
2           3   17.2   45.9       69.3    9.3
3           4  151.5   41.3       58.5   18.5
4           5  180.8   10.8       58.4   12.9

In [4]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 200 entries, 0 to 199
Data columns (total 5 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   Unnamed: 0  200 non-null    int64  
 1   TV          200 non-null    float64
 2   Radio       200 non-null    float64
 3   Newspaper   200 non-null    float64
 4   Sales       200 non-null    float64
dtypes: float64(4), int64(1)
memory usage: 7.9 KB


<font color='blue'><font size=3>As per above the data is having an unnamed column which is actually serial no of each row and need to be dropped.
There are a total of 200 samples.
The data is about advertising done through different media and the sales achieved.</font>

In [5]:
data.isna().sum()

Unnamed: 0    0
TV            0
Radio         0
Newspaper     0
Sales         0
dtype: int64

<font color='blue'><font size=3> There is no invalid data in the columns and no null values too

# 2. Check the summary statistics and discuss the max, min, avg, median, and percentiles.

In [6]:
data1 = data.drop('Unnamed: 0',axis=1)

<font color='blue'><font size=3>Droped the serial number column as it is not required

In [7]:
data1.describe()

TV       Radio   Newspaper       Sales
count  200.000000  200.000000  200.000000  200.000000
mean   147.042500   23.264000   30.554000   14.022500
std     85.854236   14.846809   21.778621    5.217457
min      0.700000    0.000000    0.300000    1.600000
25%     74.375000    9.975000   12.750000   10.375000
50%    149.750000   22.900000   25.750000   12.900000
75%    218.825000   36.525000   45.100000   17.400000
max    296.400000   49.600000  114.000000   27.000000

# a)minimum values of data

In [8]:
minimum = np.min(data1)
print(minimum)

TV           0.7
Radio        0.0
Newspaper    0.3
Sales        1.6
dtype: float64


<font color='blue'><font size=3>Advertisement in Radio was *nil* at some instances</font>

# b)maximum values of data

In [9]:
maximum = np.max(data1)
print(maximum)

TV           296.4
Radio         49.6
Newspaper    114.0
Sales         27.0
dtype: float64


<font color='blue'><font size=3>Advertisement in TV has *maximum* spending in one instance</font>

# c)average values of data

In [10]:
data1.mean()

TV           147.0425
Radio         23.2640
Newspaper     30.5540
Sales         14.0225
dtype: float64

<font color='blue'><font size=3>Comparitively more money is spent on advertisement through TV and less through Radio</font>

# d)median values of data

In [11]:
data1.median()

TV           149.75
Radio         22.90
Newspaper     25.75
Sales         12.90
dtype: float64

<font color='blue'><font size=3>Advertisement in TV has larger spendings compared to other media</font>

# e)percentiles values of data

In [12]:
data1.quantile([.25,.5,.75])

TV   Radio  Newspaper   Sales
0.25   74.375   9.975      12.75  10.375
0.50  149.750  22.900      25.75  12.900
0.75  218.825  36.525      45.10  17.400

<font color='blue'><font size=3>Data shows that the spread of advertisement on TV is much wider than other medias</font>

# 3.The manager wants to find out whether the same amount was spent for the three advertisements (TV , Radio and Newspaper). Comment on your findings.

<font color='green'><font size=3>Manager wants to find whether same amount was spent for three advertisements. For that we need to check whether amount spent of all modes of advts are same or not using oneway Anova
   # Hypothesis:
    H0: Amount Spent on three Advertisements are same
    H1: Amount spent on three advertisements are different

In [13]:
data1=data1.drop('Sales',axis=1)
fstatistic,pvalue=scp.f_oneway(data1['TV'],data1['Radio'],
                               data1['Newspaper'])

In [14]:
# DF within group
DFw=np.size(data1)-len(data1.columns)
#DF between Group
DFb =len(data1.columns)-1

In [15]:
fcriteria=scp.f.ppf(1-0.05,DFb,DFw)
scp.f.ppf(1-0.05,DFb,DFw)

3.010815204196057

<font color='blue'><font size=3>F criteria value for confidence level 0.05 with Df(SSB) 2 
and df(SSW) 597 is 3.0108

In [16]:
print('F score:',fstatistic)
print('P value:',pvalue)

F score: 358.8514595342597
P value: 4.552931539744962e-103


<font color='blue'><font size=3>F score for data is 358.8514 and pvalue is very small

In [17]:
if fstatistic>fcriteria and pvalue<0.05:
    print('Reject H0:Amount Spent on three Advertisements are same')
else:
    print('Reject H1:Amount Spent on three Advertisements are different')

Reject H0:Amount Spent on three Advertisements are same


<font color='red'><font size=3> As per the test outcome we **failed to Reject that Amount spent on three advertisement are different** and hence Amount spent is different for 3 advertisements